indigo_app/static/javascript/indigo/views/document_editor.js

```diff
@@ -349,13 +275,13 @@
 
       if (!this.comparisonDocumentId) return;
 
-      data.document = this.parent.model.toJSON();
+      data.document = this.document.toJSON();
       data.document.content = this.parent.documentContent.toXml();
-      data.element_id = this.parent.fragment.getAttribute('eId');
+      data.element_id = this.xmlElement.getAttribute('eId');
 
-      if (!data.element_id && this.parent.fragment.tagName !== "akomaNtoso") {
+      if (!data.element_id && this.xmlElement.tagName !== "akomaNtoso") {
         // for elements without ids (preamble, preface, components)
-        data.element_id = this.parent.fragment.tagName;
+        data.element_id = this.xmlElementxmlElement.tagName;
```

Comment:
```suggestion
        data.element_id = this.xmlElement.tagName;
```

Reply:
I know this matches the old code but a tagName is not an id?

Reply:
preamble, preface etc. aren't allowed eIds (🙄 ) because they are unique, so we have to use the element name.